.class public Lcom/virgilsecurity/crypto/pythia/PythiaException;
.super Ljava/lang/RuntimeException;
.source "PythiaException.java"


# static fields
.field public static final ERROR_BAD_ARGUMENTS:I = -0x1

.field public static final ERROR_PYTHIA_INNER_FAIL:I = -0xc8

.field public static final ERROR_RNG_FAILED:I = -0xca

.field public static final SUCCESS:I


# instance fields
.field private statusCode:I


# direct methods
.method public constructor <init>(I)V
    .locals 0
    .param p1, "statusCode"    # I

    .line 56
    invoke-direct {p0}, Ljava/lang/RuntimeException;-><init>()V

    .line 57
    iput p1, p0, Lcom/virgilsecurity/crypto/pythia/PythiaException;->statusCode:I

    .line 58
    return-void
.end method


# virtual methods
.method public getMessage()Ljava/lang/String;
    .locals 1

    .line 65
    iget v0, p0, Lcom/virgilsecurity/crypto/pythia/PythiaException;->statusCode:I

    sparse-switch v0, :sswitch_data_0

    .line 75
    const-string v0, "Unknown error"

    return-object v0

    .line 67
    :sswitch_0
    const-string v0, "No errors was occurred."

    return-object v0

    .line 69
    :sswitch_1
    const-string v0, "This error should not be returned if assertions is enabled."

    return-object v0

    .line 71
    :sswitch_2
    const-string v0, "Underlying pythia library returns -1."

    return-object v0

    .line 73
    :sswitch_3
    const-string v0, "Underlying random number generator failed."

    return-object v0

    :sswitch_data_0
    .sparse-switch
        -0xca -> :sswitch_3
        -0xc8 -> :sswitch_2
        -0x1 -> :sswitch_1
        0x0 -> :sswitch_0
    .end sparse-switch
.end method

.method public getStatusCode()I
    .locals 1

    .line 61
    iget v0, p0, Lcom/virgilsecurity/crypto/pythia/PythiaException;->statusCode:I

    return v0
.end method
